server/cluster.go

```diff
@@ -443,18 +443,51 @@ func (c *RaftCluster) SetStoreWeight(storeID uint64, leader, region float64) err
 }
 
 func (c *RaftCluster) checkStores() {
+	var offlineStores []*metapb.Store
+	var upStoreCount uint64
+
 	cluster := c.cachedCluster
-	for _, store := range cluster.getMetaStores() {
+	allStoresFull := true
+
+	for _, store := range cluster.GetStores() {
 		if store.GetState() != metapb.StoreState_Offline {
+			if !store.IsLowSpace(cluster.GetLowSpaceRatio()) {
+				allStoresFull = false
+			}
+
+			if store.GetState() == metapb.StoreState_Up {
+				upStoreCount++
+			}
 			continue
 		}
-		if c.storeIsEmpty(store.GetId()) {
-			err := c.BuryStore(store.GetId(), false)
+
+		offlineStore := store.Store
+		if c.storeIsEmpty(offlineStore.GetId()) {
+			err := c.BuryStore(offlineStore.GetId(), false)
 			if err != nil {
-				log.Errorf("bury store %v failed: %v", store, err)
+				log.Errorf("bury store %v failed: %v", offlineStore, err)
 			} else {
-				log.Infof("buried store %v", store)
+				log.Infof("buried store %v", offlineStore)
 			}
+		} else {
+			offlineStores = append(offlineStores, offlineStore)
+		}
+	}
+
+	if offlineStores == nil {
```

Comment:
It is a good habit to use `len(offlineStores) == 0` instead, compare slice to nil could lead to unexpected result in some cases.